standardguess

```diff
@@ -0,0 +1,382 @@
+#!/bin/bash
+#checks to see if a video's properties are consistent with a set of standards and if so it assigns expected values to other properties if they are undefined or differ from expectations 
+
+SCRIPTDIR=$(dirname $(which "${0}"))
+. "${SCRIPTDIR}/mmfunctions" || { echo "Missing '${SCRIPTDIR}/mmfunctions'. Exiting." ; exit 1 ;};
+
+_get_frame_rate(){
+    FRAMERATE=$(ffprobe -i "${1}" -v error -select_streams v:0 -show_entries stream=avg_frame_rate -of default=noprint_wrappers=1:nokey=0 | grep "^avg_frame_rate=" | cut -d = -f 2)
+}
+_get_format(){
+    FORMAT=$(ffprobe -i "${1}" -v error -select_streams v:0 -show_format -of default=noprint_wrappers=1:nokey=0 | grep "^format_name=" | cut -d = -f 2)
+}
+_MKV_Prop_Check(){
+
+    ELEMENT="$2"
+    EXPECTATION="$3"
+    MKVINFO_PROP=$(mkvinfo -t "${1}")
+    SELECTED_ELEMENT=$(echo "${MKVINFO_PROP}" | grep -c "+ ${ELEMENT}:")
+    SELECTED_ELEMENT_VALUE=$(echo "${MKVINFO_PROP}" | grep -i "+ ${ELEMENT}: " | cut -d: -f 2 | sed 's/ //g' | sed 's/(aspectratio)//g')
+
+    if [[ "${SELECTED_ELEMENT}" != "0" ]]; then
+        if [[ "${SELECTED_ELEMENT_VALUE}" -eq "${EXPECTATION}" ]]; then
+            echo "${ELEMENT} -eq expectation ${EXPECTATION}"
+        elif [[ "${SELECTED_ELEMENT_VALUE}" == "${EXPECTATION}" ]]; then
+            echo "${ELEMENT} == expectation ${EXPECTATION}"
+        else
+            echo "${ELEMENT} is not ${EXPECTATION} but it has the specified value of ${SELECTED_ELEMENT_VALUE}"
+        fi
+    else
+        echo "${ELEMENT} is not present"
+    fi
+}
+_MKV_Interlaced(){
+    MKVINTERLACED=$(_MKV_Prop_Check "${1}" "Interlaced" "1")
+}
+_MKV_WIDTH(){
+    MKVWIDTH=$(_MKV_Prop_Check "${1}" "Display width" "4")
+}
+_MKV_HEIGHT(){
+    MKVHEIGHT=$(_MKV_Prop_Check "${1}" "Display height" "3")
+}
+_MKV_UNIT(){
+    MKVUNIT=$(_MKV_Prop_Check "${1}" "Display unit" "3")
+}
+_MKV_RANGE(){
+    MKVRANGE=$(_MKV_Prop_Check "${1}" "Colour range" "1")
+}
+_MKV_TRANSFER(){
+    MKVTRANSFER=$(_MKV_Prop_Check "${1}" "Colour transfer" "1")
+}
+_MKV_PRIMARIES(){
+    MKVPRIMARIES=$(_MKV_Prop_Check "${1}" "Colour primaries" "6")
+}
```

Comment:
Are these above functions needed, from _MKV_Interlaced through _MKV_PRIMARIES? It seems that when they are used below the _MKV_Prop_Check version is also used, so many tests happen twice.